### ISSUE TITLE
GPU instances provisioned in the eastus region never move from the FailedScheduling state and remain unavailable

### ISSUE THREAD

**Issue by karlmutch** [Azure/acs-engine#3645]
**Is this a request for help?**:
---
Yes

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
---
Issue

**What version of acs-engine?**:
---
acs-engine version
Version: v0.19.5
GitCommit: c8feb6ce
GitTreeState: clean

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes 1.10

**What happened**:
ND6s machines in one region, eastus, have the nvidia.com/gpu resource on their k8s node resource but are stuck in the FailedScheduling state. 
```
kubectl describe pod studioml-go-runner
-deployment-5554f9d79-24z58                                                                                                                                    
Name:           studioml-go-runner-deployment-5554f9d79-24z58                                                                                                  
Namespace:      default
Node:           <none>
Labels:         app=studioml-go-runner
                pod-template-hash=111095835
Annotations:    <none>
Status:         Pending
IP:
Controlled By:  ReplicaSet/studioml-go-runner-deployment-5554f9d79
Containers:
  studioml-go-runner:
    Image:      quotaworkaround001.azurecr.io/sentient.ai/studio-go-runner/runner:0.5.0-feature-114-s3-cache-1fgZmj
    Port:       <none>
    Host Port:  <none>
    Limits:
      cpu:             4
      memory:          50Gi
      nvidia.com/gpu:  1
    Requests:
      cpu:             4
      memory:          50Gi
      nvidia.com/gpu:  1
    Environment Variables from:
      studioml-env  ConfigMap  Optional: false
    Environment:    <none>
    Mounts:
      /tmp from tmp-volume (rw)
      /usr/lib/x86_64-linux-gnu/libcuda.so.1 from libcuda (rw)
      /usr/local/nvidia from nvidia (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-5rkb9 (ro)
Conditions:
  Type           Status
  PodScheduled   False
Volumes:
  tmp-volume:
    Type:    EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
  nvidia:
    Type:          HostPath (bare host directory volume)
    Path:          /usr/local/nvidia
    HostPathType:
  libcuda:
    Type:          HostPath (bare host directory volume)
    Path:          /usr/lib/x86_64-linux-gnu/libcuda.so.1
    HostPathType:
  default-token-5rkb9:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-5rkb9
    Optional:    false
QoS Class:       Guaranteed
Node-Selectors:  beta.kubernetes.io/os=linux
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
                 nvidia.com/gpu:NoSchedule
Events:
  Type     Reason            Age                 From               Message
  ----     ------            ----                ----               -------
  Warning  FailedScheduling  1m (x107 over 32m)  default-scheduler  0/31 nodes are available: 1 Insufficient cpu, 1 Insufficient memory, 31 Insufficient nvidia.com/gpu.
```
**What you expected to happen**:
I expected that the k8s node would eventually be schedule and available for pods.

**How to reproduce it** (as minimally and precisely as possible):
Use acs-engine to create a cluster in the eastus region, using the following json file as your api-model
```
{
  "apiVersion": "vlabs",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
      "orchestratorRelease": "1.10"
    },
    "masterProfile": {
      "count": 1,
      "dnsPrefix": "",
      "vmSize": "Standard_D2_v2"
    },
    "agentPoolProfiles": [
      {
        "name": "agentpool1",
        "count": 1,
        "vmSize": "Standard_ND6s",
        "OSDiskSizeGB": 200,
        "availabilityProfile": "VirtualMachineScaleSets"
      }
    ],
    "linuxProfile": {
      "adminUsername": "azureuser",
      "ssh": {
        "publicKeys": [
          {
            "keyData": ""
          }
        ]
      }
    },
    "servicePrincipalProfile": {
     "clientId": "",
     "secret": ""
    },
    "certificateProfile": {}
  }
}
```
```
acs-engine deploy --resource-group $k8s_resource_group --subscription-id $subscription_id --dns-prefix $k8s_resource_group --location eastus --auto-suffix --output-directory test_gpu --api-model [file above]
```
Once everything completes use the kubectl describe node command and observe that there is an errors even after waiting for a long period of time.

```
Events:
  Type     Reason            Age                 From               Message
  ----     ------            ----                ----               -------
  Warning  FailedScheduling  1m (x107 over 32m)  default-scheduler  0/31 nodes are available: 1 Insufficient cpu, 1 Insufficient memory, 31 Insufficient nvidia.com/gpu.
```

**Anything else we need to know**:
A ticket exists in the MSFT support ticketing system, 118080918759525

**Comment by CecileRobertMichon** [comment]
Related to https://github.com/Azure/acs-engine/issues/3603

**Comment by CecileRobertMichon** [comment]
@karlmutch looks like this was fixed in [0.20.0](https://github.com/Azure/acs-engine/releases/tag/v0.20.0)
